In [1]:
import pandas as pd
from pyjstat import pyjstat
pd.set_option('display.max_colwidth', 500)
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

<b> I have been provided with following two urls to determine in which new countries Futurice can expand its location
<li> size of ICT sector in countries 
<li> amount of cloud computing used in the country
<p><b> Additional GDP data has been provided </b></p>

In [2]:
#url json stat
url_ict = 'http://ec.europa.eu/eurostat/wdds/rest/data/v2.1/json/en/tin00074?nace_r2=ICT'
url_percentage_cloud_computing = 'http://ec.europa.eu/eurostat/wdds/rest/data/v2.1/json/en/isoc_cicce_use?sizen_r2=M_C10_S951_XK&sizen_r2=L_C10_S951_XK&unit=PC_ENT&indic_is=E_CC'

# Percentage of the ICT sector on GDP

In [9]:
# ict data
dataset_ict = pyjstat.Dataset.read(url_ict)

In [5]:
print(dataset_ict)

Dataset([('version', '2.0'), ('label', 'Percentage of the ICT sector on GDP'), ('href', 'http://ec.europa.eu/eurostat/wdds/rest/data/v2.1/json/en/tin00074?nace_r2=ICT'), ('source', 'Eurostat'), ('updated', '2022-07-01'), ('status', {'0': 'b', '6': ':c', '10': 'b', '12': ':b', '13': ':', '14': ':', '15': ':', '16': ':', '17': ':', '19': 'e', '20': 'e', '21': ':c', '22': ':c', '23': ':c', '24': ':c', '34': 'b', '36': 'b', '46': 'b', '48': ':b', '49': ':', '50': ':', '51': ':', '52': ':', '53': ':', '54': ':', '55': ':', '56': ':', '57': ':', '58': ':', '59': ':', '60': ':c', '61': ':c', '62': ':c', '63': ':c', '64': ':c', '65': ':c', '66': ':c', '67': ':c', '68': ':c', '69': ':c', '70': ':c', '71': ':c', '72': ':c', '84': 'b', '94': 'bp', '95': 'p', '96': 'b', '100': ':c', '101': ':c', '102': ':c', '103': ':c', '104': ':c', '105': ':c', '106': 'b', '108': 'b', '118': 'b', '120': 'b', '121': ':c', '122': ':c', '126': 'bp', '127': 'p', '128': 'p', '129': 'p', '130': 'b', '131': 'p', '132':

<b>The response from the json stat is a Dataset.
The data is in tabular format (cube)
<br>
Listing few properties of the dataset:
<li> Version is 2.0
<li> label : Percentage of the ICT sector on GDP
<li> source : Eurostat
<li> updated : 2022-07-01
<li> status : {}  ( 'b': 'break in time series', ':': 'not available', 'c': 'confidential', 'e': 'estimated', 'p': 'provisional' )
<li> class : dataset
<li> dimension : {nace_r2, geo , time}
 

In [11]:
# writing dataset to pandas dataframe
df = dataset_ict.write('dataframe')

In [4]:
df.head()

nace_r2      geo  time  value
0  Information and Communication Technology - Total  Austria  2008   3.19
1  Information and Communication Technology - Total  Austria  2009   3.14
2  Information and Communication Technology - Total  Austria  2010   3.12
3  Information and Communication Technology - Total  Austria  2011   3.25
4  Information and Communication Technology - Total  Austria  2012   3.15

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 444 entries, 0 to 443
Data columns (total 4 columns):
nace_r2    444 non-null object
geo        444 non-null object
time       444 non-null object
value      270 non-null float64
dtypes: float64(1), object(3)
memory usage: 14.0+ KB


###### From the above we can see that values are missing for 174 geo time combinations

In [25]:
# for following fields the values are null
df.drop('nace_r2', axis = 1, inplace = True)
print(df[df['value'].isna()])

                                           geo  time  value
6                                      Austria  2014    NaN
12                      Bosnia and Herzegovina  2008    NaN
13                      Bosnia and Herzegovina  2009    NaN
14                      Bosnia and Herzegovina  2010    NaN
15                      Bosnia and Herzegovina  2011    NaN
16                      Bosnia and Herzegovina  2012    NaN
17                      Bosnia and Herzegovina  2013    NaN
21                      Bosnia and Herzegovina  2017    NaN
22                      Bosnia and Herzegovina  2018    NaN
23                      Bosnia and Herzegovina  2019    NaN
24                                     Belgium  2008    NaN
48                                 Switzerland  2008    NaN
49                                 Switzerland  2009    NaN
50                                 Switzerland  2010    NaN
51                                 Switzerland  2011    NaN
52                                 Switz

In [12]:
print(df['time'].unique())

['2008' '2009' '2010' '2011' '2012' '2013' '2014' '2015' '2016' '2017'
 '2018' '2019']


###### The values for ICT are  from 2008 until 2019

In [13]:
print(df['geo'].unique())

['Austria' 'Bosnia and Herzegovina' 'Belgium' 'Bulgaria' 'Switzerland'
 'Cyprus' 'Czechia' 'Germany (until 1990 former territory of the FRG)'
 'Denmark' 'Estonia' 'Greece' 'Spain'
 'European Union - 27 countries (from 2020)' 'Finland' 'France' 'Croatia'
 'Hungary' 'Ireland' 'Iceland' 'Italy' 'Lithuania' 'Luxembourg' 'Latvia'
 'Montenegro' 'North Macedonia' 'Malta' 'Netherlands' 'Norway' 'Poland'
 'Portugal' 'Romania' 'Serbia' 'Sweden' 'Slovenia' 'Slovakia' 'Turkey'
 'United Kingdom']


<p><b>ICT for above list of countries are available.</b></p><br>
<b>There are also aggregates e.g European Union - 27 countries (from 2020)

# % of Enterprises in cloud computing by country euro stat data

In [27]:
url_percentage_cloud_computing = 'http://ec.europa.eu/eurostat/wdds/rest/data/v2.1/json/en/isoc_cicce_use?sizen_r2=M_C10_S951_XK&sizen_r2=L_C10_S951_XK&unit=PC_ENT&indic_is=E_CC'
dataset_percentage_cloud_computing = pyjstat.Dataset.read(url_percentage_cloud_computing)

In [28]:
dataset_percentage_cloud_computing

Dataset([('version', '2.0'),
         ('label', 'Cloud computing services'),
         ('href',
          'http://ec.europa.eu/eurostat/wdds/rest/data/v2.1/json/en/isoc_cicce_use?sizen_r2=M_C10_S951_XK&sizen_r2=L_C10_S951_XK&unit=PC_ENT&indic_is=E_CC'),
         ('source', 'Eurostat'),
         ('updated', '2022-03-17'),
         ('status',
          {'0': ':',
           '1': ':',
           '2': ':',
           '3': ':',
           '4': ':',
           '5': ':',
           '8': ':',
           '14': ':',
           '15': ':',
           '16': ':',
           '17': ':',
           '43': ':',
           '50': ':',
           '52': ':',
           '64': ':',
           '66': ':',
           '71': ':',
           '73': ':',
           '92': ':',
           '94': ':',
           '96': ':',
           '97': ':',
           '99': ':',
           '101': ':',
           '103': ':',
           '104': ':',
           '106': ':',
           '108': ':',
           '113': ':',
           '115': ':'

<b>The response from the json stat is a Dataset.
The data is in tabular format (cube)
<br>
Listing few properties of the dataset:
<li> Version is 2.0
<li> label : Cloud computing services
<li> source : Eurostat
<li> updated : 2022-07-01
<li> status : {}  ( 'b': 'break in time series', ':': 'not available', 'c': 'confidential', 'e': 'estimated', 'p': 'provisional' )
<li> class : dataset
<li> dimension : {'unit', 'sizen_r2', 'indic_is', 'geo', 'time'}

In [30]:
# write to pandas dataframe
df_cc = dataset_percentage_cloud_computing.write('dataframe')

In [32]:
df_cc.head()

unit  \
0  Percentage of enterprises   
1  Percentage of enterprises   
2  Percentage of enterprises   
3  Percentage of enterprises   
4  Percentage of enterprises   

                                                                                        sizen_r2  \
0  Large enterprises (250 employees and self-employed persons or more), without financial sector   
1  Large enterprises (250 employees and self-employed persons or more), without financial sector   
2  Large enterprises (250 employees and self-employed persons or more), without financial sector   
3  Large enterprises (250 employees and self-employed persons or more), without financial sector   
4  Large enterprises (250 employees and self-employed persons or more), without financial sector   

                                              indic_is      geo  time  value  
0  Buy cloud computing services used over the internet  Albania  2014    NaN  
1  Buy cloud computing services used over the internet  Albania  2015    NaN  
2  Buy cloud computing services used over the internet  Albania  2016    NaN  
3  Buy cloud computing services used over the internet  Albania  2017    NaN  
4  Buy cloud computing services used over the internet  Albania  2018    NaN

<b>Following attributes are present:
<li> unit: Percentage of enterprises
<li> Sizen_r2 : Large enterprises (250 employees and self-employed persons or more), without financial sector
<li> indic_is : Buy cloud computing services used over the internet	
<li> geo : countries
<li> time : year
<li> value : percentage values

In [33]:
print(df_cc['time'].unique())

['2014' '2015' '2016' '2017' '2018' '2020' '2021']


<b>From year 2014 until 2021 percentages of enterprises in Cloud computing in different european countries are present


In [34]:
print(df_cc['geo'].unique())

['Albania' 'Austria' 'Bosnia and Herzegovina' 'Belgium' 'Bulgaria'
 'Cyprus' 'Czechia' 'Germany (until 1990 former territory of the FRG)'
 'Denmark'
 'Euro area (EA11-1999, EA12-2001, EA13-2007, EA15-2008, EA16-2009, EA17-2011, EA18-2014, EA19-2015)'
 'Estonia' 'Greece' 'Spain' 'European Union - 15 countries (1995-2004)'
 'European Union - 27 countries (2007-2013)'
 'European Union - 27 countries (from 2020)'
 'European Union - 28 countries (2013-2020)' 'Finland' 'France' 'Croatia'
 'Hungary' 'Ireland' 'Iceland' 'Italy' 'Lithuania' 'Luxembourg' 'Latvia'
 'Montenegro' 'North Macedonia' 'Malta' 'Netherlands' 'Norway' 'Poland'
 'Portugal' 'Romania' 'Serbia' 'Sweden' 'Slovenia' 'Slovakia' 'Turkey'
 'United Kingdom']


<b> We can see the above list of countries are considered in the dataset </b>
<br> <b>Also there are some stats aggregated by Euro area, some by countries in European Union</b>

In [35]:
df_cc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 574 entries, 0 to 573
Data columns (total 6 columns):
unit        574 non-null object
sizen_r2    574 non-null object
indic_is    574 non-null object
geo         574 non-null object
time        574 non-null object
value       442 non-null float64
dtypes: float64(1), object(5)
memory usage: 27.0+ KB


<b>Values are missing for 132 combinations of geo, time

# GDP data

In [36]:
df_gdp = pd.read_csv('gdp_data.csv', header='infer', delimiter='|', decimal=',')

In [37]:
df_gdp.head()

Country     2008     2012     2014     2016     2018  \
0                 Austria      NaN  318.653  333.146  357.300  385.712   
1                 Belgium  351,743  386.175  403.003      NaN  459.820   
2  Bosnia and Herzegovina   18,640      NaN   18.560      NaN   19.900   
3                Bulgaria   37,218   42.034      NaN   48.621   56.087   
4                 Croatia   47,998   43.941   43.401   46.616   51.625   

      2019  
0  397.575  
1  476.203  
2      NaN  
3   61.239  
4   54.237

In [38]:
df_gdp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 7 columns):
Country    29 non-null object
2008       27 non-null object
2012       22 non-null float64
2014       21 non-null float64
2016       20 non-null float64
2018       20 non-null float64
2019       21 non-null float64
dtypes: float64(5), object(2)
memory usage: 1.7+ KB


<b>There are values missing over the years
<li> Also for 2008 the values seems to be not clean, whichever country futurice is present, those text seems to have moved to this column

In [39]:
(df_gdp['Country'].unique())

array(['Austria', 'Belgium', 'Bosnia and Herzegovina', 'Bulgaria',
       'Croatia', 'Czechia', 'Denmark', 'Estonia', 'Finland', 'France',
       'Germany', 'Greece', 'Hungary', 'Iceland', 'Italy', 'Latvia',
       'Lithuania', 'Malta', 'Netherlands', 'North Macedonia', 'Poland',
       'Portugal', 'Romania', 'Serbia', 'Slovakia', 'Slovenia', 'Spain',
       'Sweden', 'United Kingdom'], dtype=object)

<b>There is data for 29 countries

In [40]:
(df_gdp['2008'].unique())

array([nan, '351,743', '18,640', '37,218', '47,998', '161,313', '241,614',
       '16,638', 'Offices in Helsinki and Tampere', '1992,38',
       'Offices in Berlin, Munich and Stuttgart', '241,99', '108,216',
       '13,4', '1637,70', '24,397', '32,696', '6,129', '647,198', '9,1',
       '366,181', '179,103', '146,591', '66,098', '37,926', '1109,54',
       'Office in Stockholm', 'Office in London'], dtype=object)

<b> The column 2008 has the string values in which countries Futurice is already present